### PR TITLE
shields: add WAVESHARE e-Paper shield

### PR DESCRIPTION
**Pull request by jfischer-no**
Add WAVESHARE e-Paper shield and support for
GDEH0213B1, GDEH0213B72 and GDEH029A1 displays.

### CODE REVIEW

**Review by vanwinkeljan — APPROVED** [review]

File: boards/shields/waveshare_epaper/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+WAVESHARE e-Paper Raw Panel Shield
+##################################
+
+Overview
+********
+
+The WAVESHARE e-Paper Raw Panel Shield is a universal driver shield.
+The shield can be used to drive various Electrophoretic (electronic ink)
+Display (EPD) with a SPI interface.
+This shield comes also with a 23LC1024 SPI RAM
```
Comment (dbkinder) [inline boards/shields/waveshare_epaper/doc/index.rst]: Change to:
```
This shield includes a 23LC1024 1Mb SPI Serial SRAM that is
not currently supported by the Zephyr RTOS.
```

File: boards/shields/waveshare_epaper/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
++--------------+-----------------+--------------+------------------------------+
+| Good Display | E029A01         | IL3820 /     | waveshare_epaper_gdeh029a1   |
+| GDEH029A1    |                 | ssd16xx      |                              |
++--------------+-----------------+--------------+------------------------------+
+
+
+Requirements
+************
+
+This shield can only be used with a board which provides a configuration
```
Comment (dbkinder) [inline boards/shields/waveshare_epaper/doc/index.rst]: change "which" to "that"

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for the docs.  A few suggested changes.

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks!

**Review by galak — COMMENTED** [review]
Do we have any sample or something that builds with this shield?

**Comment by jfischer-phytec-iot** [comment]
>Do we have any sample or something that builds with this shield?

@galak I have to add something like 183a95c132aa9d9d9f1ac9b0befbbb43640c25f3. I am just not sure how. There are (currently) three variants of this shield, should we test all variants or just one?

**Comment by galak** [comment]
> @galak I have to add something like [183a95c](https://github.com/zephyrproject-rtos/zephyr/commit/183a95c132aa9d9d9f1ac9b0befbbb43640c25f3). I am just not sure how. There are (currently) three variants of this shield, should we test all variants or just one?

Would be good to test all 3, just so we at least have one build test for each shield.

**Review by erwango — APPROVED** [review]

**Pull request merged**